lib/metrics/helpers/processMetricsHelpers.js

```diff
@@ -3,11 +3,13 @@
 function aggregateByObjectName(list) {
 	const data = {};
 
-	for (const key in list) {
-		if (data.hasOwnProperty(list[key].constructor.name)) {
-			data[list[key].constructor.name] += 1;
-		} else {
-			data[list[key].constructor.name] = 1;
+	for (let i = 0; i < list.length; i++) {
+		if (list[i] !== null && typeof list[i]) {
```

Comment:
what does the `typeof` do?

Reply:
What about

```js
for (const key in list) {
	const listElement = list[key];

	if (!listElement) {
		continue;
	}

	if (data.hasOwnProperty(listElement.constructor.name)) {
		data[listElement.constructor.name] += 1;
	} else {
		data[listElement.constructor.name] = 1;
	}
}
```

?

Reply:
@SimenB you are right - this check for "undefined" will never works.
I've rewrote this code as you proposed. New commit squashed with original one